### PR TITLE
use proper intra doc link

### PR DESCRIPTION
**Pull request by dharjeezy**
Use proper intra doc link  as mentioned in issue #9962 @thiolliere 

Polkadot address:
12ZNas89oEagaxLVNbpqmvfMxdrGrqN7gJKSpwthTUPZsrku

### CODE REVIEW

File: frame/support/procedural/src/construct_runtime/expand/origin.rs [hunk 1]
```diff
@@ -182,15 +182,15 @@ pub fn expand_outer_origin(
 		// For backwards compatibility and ease of accessing these functions.
 		#[allow(dead_code)]
 		impl Origin {
-			/// Create with system none origin and `frame-system::Config::BaseCallFilter`.
+			/// Create with system none origin and [`frame_system::Config::BaseCallFilter`].
```
Comment (bkchr) [inline frame/support/procedural/src/construct_runtime/expand/origin.rs:185]: You need to use `#system_path`

**Review by bkchr — CHANGES_REQUESTED** [review]

File: frame/support/procedural/src/construct_runtime/expand/origin.rs (hunk 1 above)
Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:185]: Oh... Can you elaborate more what you mean by system_path?

Comment (bkchr) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:185]: @thiolliere already has written this for you: https://github.com/paritytech/substrate/issues/9962#issuecomment-949419511

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:185]: I actually tried out what @thiolliere suggested, it wasn't outputting the right docs and I ran into errors too

Comment (bkchr) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:185]: Then you need to fix this.

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:185]: Oh... OK!
Will research on how to fix that.

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: I have tried modifying what i did earlier @bkchr 
Can you take a look?

Comment (thiolliere) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: it doesn't work for me, did you tested your PR manually before requesting review ?
PR

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: my bad. @thiolliere will push a fix for this

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: @thiolliere  i have fix this in my latest push. You can check now

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: Still awaiting your reviews @thiolliere @bkchr

Comment (thiolliere) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: we will review when we get time

File: frame/support/procedural/src/construct_runtime/expand/origin.rs [hunk 2]
```diff
@@ -68,14 +68,39 @@ pub fn expand_outer_origin(
...
+		get_intra_doc_string(" Create with system none origin and", &system_path_name);
+
+	let doc_string_root_origin =
+		get_intra_doc_string(" Create with system root origin and", &system_path_name);
+
+	let doc_string_signed_origin =
+		get_intra_doc_string(" Create with system signed origin and", &system_path_name);
+
+	let doc_string_runtime_origin =
+		get_intra_doc_string(" Convert to runtime origin, using as filter and", &system_path_name);
```
Comment (thiolliere) [inline frame/support/procedural/src/construct_runtime/expand/origin.rs:91]:
```suggestion
		get_intra_doc_string(" Convert to runtime origin, using as filter:", &system_path_name);
```

**Review by thiolliere — APPROVED** [review]
looks good to me

File: frame/support/procedural/src/construct_runtime/expand/origin.rs [hunk 3]
```diff
@@ -68,14 +68,39 @@ pub fn expand_outer_origin(
 	}
 
 	let system_path = &system_pallet.path;
+
 	let system_index = system_pallet.index;
 
+	let system_path_name = system_path.module_name();
+
+	let doc_string = get_intra_doc_string(
+		" Origin is always created with the base filter configured in",
```
Comment (bkchr) [inline frame/support/procedural/src/construct_runtime/expand/origin.rs:77]: Can we please get rid of the leading whitespaces in all these codes.

**Review by bkchr — APPROVED** [review]
One nitpick, otherwise looks good.

File: frame/support/procedural/src/construct_runtime/expand/origin.rs (hunk 3 above)
Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:77]: Alright. Will do

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: @bkchr i have done that

Comment (bkchr) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs]: Ty

File: frame/support/procedural/src/construct_runtime/expand/origin.rs [hunk 4]
```diff
@@ -68,14 +68,39 @@ pub fn expand_outer_origin(
 	}
 
 	let system_path = &system_pallet.path;
+
 	let system_index = system_pallet.index;
 
+	let system_path_name = system_path.module_name();
+
+	let doc_string = get_intra_doc_string(
+		"Origin is always created with the base filter configured in",
```
Comment (bkchr) [inline frame/support/procedural/src/construct_runtime/expand/origin.rs:77]: One last question, so this leading whitespace was never needed? I would have thought that you may need to add it in your function below, but there is already one.

I just want to have this clarified and then I'm going to merge this.

Comment (dharjeezy) [reply frame/support/procedural/src/construct_runtime/expand/origin.rs:77]: Yes... The leading whitespace was never needed. You can go ahead and merge

**Comment by dharjeezy** [comment]
I have added my Polkadot address correctly now.
Can you try tipping again.
Thanks @bkchr

**Pull request merged**